ulib/experimental/Steel.Hashtbl.fsti

```diff
@@ -0,0 +1,100 @@
+(*
+   Copyright 2021 Microsoft Research
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+   Authors: Aseem Rastogi
+*)
+
+module Steel.Hashtbl
+
+open Steel.Memory
+open Steel.Effect
+open Steel.Effect.Atomic
+
+module G = FStar.Ghost
+module Set = FStar.Set
+module Map = FStar.Map
+module U32 = FStar.UInt32
+
+/// This module provides a basic hashtable with no support for hash-collisions
+///
+/// In particular, if the table already contains a key k1 and the client invokes `put` API
+///   with another key k2 that has a hash-collision with k1, then k1 is removed from the table
```

Comment:
It would be good to mention in this summary comment the ghost behavior as well. 

i.e., that logically, the module provides a map, but only a subset of that map is maintained in concrete state. So, putting k2 in the scenario above changes the concrete state but leaves the ghost view unchanged for k1.